app/src/main/java/com/keylesspalace/tusky/db/DraftsAlert.kt

```diff
@@ -37,51 +39,55 @@ import kotlinx.coroutines.launch
 
 private const val TAG = "DraftsAlert"
 
-@Singleton
 class DraftsAlert @Inject constructor(
     db: AppDatabase,
     private val accountManager: AccountManager
 ) {
     // For tracking when a media upload fails in the service
     private val draftDao: DraftDao = db.draftDao()
 
+    private var dialog: AlertDialog? = null
+
     fun <T> observeInContext(context: T, showAlert: Boolean) where T : Context, T : LifecycleOwner {
         accountManager.activeAccount?.let { activeAccount ->
             val coroutineScope = context.lifecycleScope
 
-            // Assume a single MainActivity, AccountActivity or DraftsActivity never sees more then one user id in its lifetime.
+            // One activity never sees more then one user id in its lifetime.
             val activeAccountId = activeAccount.id
 
-            val draftsNeedUserAlert = draftDao.draftsNeedUserAlert(activeAccountId)
-
             // observe ensures that this gets called at the most appropriate moment wrt the context lifecycle—
             // at init, at next onResume, or immediately if the context is resumed already.
             coroutineScope.launch {
-                if (showAlert) {
-                    draftsNeedUserAlert.collect { count ->
-                        Log.d(TAG, "User id $activeAccountId changed: Notification-worthy draft count $count")
-                        if (count > 0) {
-                            MaterialAlertDialogBuilder(context)
-                                .setTitle(R.string.action_post_failed)
-                                .setMessage(
-                                    context.resources.getQuantityString(R.plurals.action_post_failed_detail, count)
-                                )
-                                .setPositiveButton(R.string.action_post_failed_show_drafts) { _: DialogInterface?, _: Int ->
-                                    clearDraftsAlert(coroutineScope, activeAccountId) // User looked at drafts
+                context.repeatOnLifecycle(Lifecycle.State.RESUMED) {
+                    val draftsNeedUserAlert = draftDao.draftsNeedUserAlert(activeAccountId)
+
+                    if (showAlert) {
+                        draftDao.draftsNeedUserAlert(activeAccountId).collect { count ->
+                            Log.d(TAG, "User id $activeAccountId changed: Notification-worthy draft count $count")
```

Comment:
What's notification-worthy?

The meaning of this log (and thus this code) is "the user changed something and thus his draft count now is $count"?

Reply:
Its about the count of drafts with the `failedToSendNew` flag